src/memory_targets/memory_target_numa.c

```diff
@@ -32,27 +34,81 @@ static enum umf_result_t numa_initialize(void *params, void **mem_target) {
 
 static void numa_finalize(void *mem_target) { free(mem_target); }
 
-static enum umf_result_t numa_pool_create_from_memspace(
-    umf_memspace_handle_t memspace, void **mem_targets,
-    umf_memspace_policy_handle_t policy, umf_memory_pool_handle_t *pool) {
-    // TODO: cast mem_targets to (struct numa_memory_target_t**), parse ids and initialize os provider
-    (void)memspace;
-    (void)mem_targets;
-    (void)policy;
-    (void)pool;
-    return UMF_RESULT_ERROR_NOT_SUPPORTED;
-}
+static const umf_os_memory_provider_params_t
+    UMF_OS_MEMORY_PROVIDER_PARAMS_DEFAULT = {
+        // Visibility & protection
+        .protection = UMF_PROTECTION_READ | UMF_PROTECTION_WRITE,
+        .visibility = UMF_VISIBILITY_PRIVATE,
+
+        // NUMA config
+        .nodemask = NULL,
+        .maxnode = 0, // TODO: numa_max_node/GetNumaHighestNodeNumber
+        .numa_mode = UMF_NUMA_MODE_DEFAULT,
+        .numa_flags = UMF_NUMA_FLAGS_STRICT, // TODO: determine default behavior
+
+        // Logging
+        .traces = 0, // TODO: parse env variable for log level?
+};
 
 static enum umf_result_t numa_memory_provider_create_from_memspace(
-    umf_memspace_handle_t memspace, void **mem_targets,
+    umf_memspace_handle_t memspace, void **mem_targets, size_t num_targets,
     umf_memspace_policy_handle_t policy,
     umf_memory_provider_handle_t *provider) {
-    // TODO: cast mem_targets to (struct numa_memory_target_t**), parse ids and initialize os provider
+    // TODO: validate that provided targets belong to the memspace?
     (void)memspace;
-    (void)mem_targets;
+    // TODO: apply policy
     (void)policy;
-    (void)provider;
-    return UMF_RESULT_ERROR_NOT_SUPPORTED;
+
+    // TODO: retrieve targets from mespace instead of passing as a param?
```

Comment:
To do this, I think we would need to have something like: https://github.com/oneapi-src/unified-runtime/pull/446 Without this, memory_target would need to know how `umf_memory_target_t` is implemented and we don't want to expose that.